notebook/notebookapp.py

```diff
@@ -608,6 +608,9 @@ def _webapp_settings_changed(self, name, old, new):
             help="Supply overrides for the tornado.web.Application that the "
                  "Jupyter notebook uses.")
     
+    terminado_settings = Dict(config=True,
+            help='Supply overrides for terminado. Currently only supports "shell_command".')
```

Comment:
It's a reasonable enough approach. I think tornado_settings is a bit of a special case because there are so many things that might go in there, but I'd be happy either way. Let's give others a chance to look at this.